lib/spack/external/altgraph/__init__.py

```diff
@@ -140,8 +140,8 @@
 
 '''
 import pkg_resources
-__version__ = pkg_resources.require('altgraph')[0].version
-
+#__version__ = pkg_resources.require('altgraph')[0].version
```

Comment:
This was needed to allow altgraph to be called from macholib. Because pkg_resources is not vendored as well I took this shortcut.

Reply:
I'm not sure I understand: we may not vendor `pkg_resources` but it would have to be available for the above `import pkg_resources` line to work. Usually installing setuptools is sufficient to allow this to work. Are you by chance developing with a Spack-built instance of python?

Reply:
Possibly at the time. I will test with system python3 since spack seems to prefer it even if it is the installed version.

Reply:
```
spack python
Spack version 0.13.2
Python 3.6.8, Linux x86_64
>>> import altlib
Traceback (most recent call last):
  File "<console>", line 1, in <module>
ModuleNotFoundError: No module named 'altlib'
>>> import altgraph
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/build/test/spack/lib/spack/external/altgraph/__init__.py", line 143, in <module>
    __version__ = pkg_resources.require('altgraph')[0].version
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 888, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 774, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'altgraph' distribution was not found and is required by the application
```

Reply:
```
>>> import macholib.MachOStandalone
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/build/test/spack/lib/spack/external/macholib/MachOStandalone.py", line 3, in <module>
    from macholib.MachOGraph import MachOGraph, MissingMachO
  File "/build/test/spack/lib/spack/external/macholib/MachOGraph.py", line 8, in <module>
    from altgraph.ObjectGraph import ObjectGraph
  File "/build/test/spack/lib/spack/external/altgraph/__init__.py", line 143, in <module>
    __version__ = pkg_resources.require('altgraph')[0].version
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 981, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 867, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'altgraph' distribution was not found and is required by the application
```

Reply:
It looks like the import path isn't set.

Reply:
I see `lib/spack/external` in `sys.path` so I need to investigate why the libraries there would not be included. This is failing for Spack-built python 3.7 and succeeding for system (MacOS) Python 2.7.

Reply:
I ended up not using MachOStandalone.
I could remove altgraph and remove the files in macholib that make use of it
MachOStandalone imports MachOFGraph imports altgraph

Reply:
If it involves a clean removal of the libraries, that works. If it involves stripping out or editing sources such that we have something that significantly differs from an available version, then I'd prefer not to. Either way, I should still determine why this import isn't working for Spack-built python.

Reply:
This may actually be expected (and your change may be the way to handle the issue): the `pkg_resources` code is not finding the altgraph import despite the fact that it is in `sys.path` I think that's because there is no `.dist-info` or `.egg-info` and because this call occurs in `altgraph/__init__.py`.

Reply:
I looked at __init__.py in macholib and __version__ is defined as the version string.